numpy/_core/arrayprint.py

```diff
@@ -1565,12 +1552,12 @@ def _array_repr_implementation(
         arr, max_line_width=None, precision=None, suppress_small=None,
         array2string=array2string):
     """Internal version of array_repr() that allows overriding array2string."""
-    override_repr = _format_options["override_repr"]
+    override_repr = _format_options.get()["override_repr"]
```

Comment:
I think here I would define local variable `_current_options = _format_options.get()` and just use that through-out, to avoid the multiple `.get()` (to avoid distraction, not for speed)

(Of course, if you go with my suggested name change, this could become `_format_options = format_options.get()` and the code below could be used unchanged.)

Reply:
You don't need an underscore inside the function :).  `curr_options = format_options.get()` seems fair for multiple lookups.

(But this is bike-shedding, and I don't care.)